### PR TITLE
Webhooks documentation.

### PR DESCRIPTION
**Pull request by roeierez**

### CODE REVIEW

**Comment by roeierez** [comment]
opening for early review.

**Review by dangeross — COMMENTED** [review]
Reads well, easy to understand

File: src/guide/payment_notification.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Receiving Payment Notifications via Webhook in Breez SDK
+
+## Overview
+The Breez SDK provides a robust feature for applications to receive real-time notifications for incoming payments over the Lightning Network. This is achieved through the use of webhooks. A webhook is a powerful tool that allows your application to be notified via a specified URL when a payment is received.
```
Comment (JssDWt) [inline src/guide/payment_notification.md:4]:
```suggestion
The Breez SDK provides a robust feature for applications to receive real-time notifications for incoming payments over the Lightning Network. This is achieved through the use of webhooks. A webhook is a powerful tool that allows your application to be notified via a specified URL when a payment is about to be received.
```

File: src/guide/payment_notification.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  "payment_hash" : [payment hash]
+ }
+}
+</pre>
+</section>
+
+## Processing the Payload
+Processing the Payload is according to the application needs. One of the major use cases that can be handled via webhooks is receiving a payment in mobile app while the app is not running, or in short "offline receive"
+
+## Webhook integration for "offline recieve"
```
Comment (JssDWt) [inline src/guide/payment_notification.md:45]:
```suggestion
## Webhook integration for "offline receive"
```

File: src/guide/payment_notification.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+In the context of mobile applications using the Breez SDK for Lightning Network payments, a crucial architectural component involves the seamless flow of payment notifications to the user's mobile device. This process is facilitated through a Notification Delivery Service (NDS), acting as an intermediary. When a payment is made to a user, the LSP sends a notification to the NDS, configured with a specific webhook URL. The NDS then processes this information and dispatches a push notification to the intended mobile device, ensuring the user receives timely updates about incoming payments. This architecture necessitates vendors to set up and maintain their own NDS, tailored to handle and forward these notifications efficiently.
+
+## Step 1: Run your own NDS
```
Comment (JssDWt) [inline src/guide/payment_notification.md:48]: Step 1 is also defined above, but they are not seperated by a header. I think these steps should start with 3 hashtags `###` and there should be a subheader seperating the sections.

File: src/guide/payment_notification.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+#### Overview
+For iOS, the app must use NotificationServiceExtension to process notifications in the background.
+
+#### Steps for Implementation
+* Integrate NotificationServiceExtension: Add NotificationServiceExtension to your iOS project. This extension allows your app to process the incoming push notification data in the background before presenting it to the user.
+
+* Wake Up the App: When a push notification is received, the NotificationServiceExtension will be triggered, waking up the app.
+
+* Connect with Breez SDK: Inside the extension, implement the logic to establish a connection with the Breez SDK which also runs the signer so the payment can be processed.
```
Comment (JssDWt) [inline src/guide/payment_notification.md:89]: Note: This is the first mention of the signer in the documentation. We should introduce this entire concept of having to be online in order to receive a payment in the documentation elsewhere.

**Review by JssDWt — COMMENTED** [review]
It looks good and seems clear.
Some points in the comments below.

File: src/SUMMARY.md [hunk 5]
```diff
@@ -21,4 +21,5 @@
 - [Buy Bitcoin](guide/buy_btc.md)
 - [Static Channel Backup](guide/static_channel_backup.md)
 - [Service Status](guide/service_status.md)
+- [Payment Notificaions](guide/payment_notification.md)
```
[the remark this answers is not in the document]
Comment (roeierez) [reply src/SUMMARY.md]: Done

File: src/guide/payment_notification.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+ "type": "payment_received",
+ "data": {  
+  "payment_hash" : [payment hash]
+ }
+}
+</pre>
+</section>
+
+## Processing the Payload
+Processing the Payload is according to the application needs. One of the major use cases that can be handled via webhooks is receiving a payment in mobile app while the app is not running, or in short "offline receive"
```
[the remark this answers is not in the document]
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+## Webhook integration for "offline recieve"
+In the context of mobile applications using the Breez SDK for Lightning Network payments, a crucial architectural component involves the seamless flow of payment notifications to the user's mobile device. This process is facilitated through a Notification Delivery Service (NDS), acting as an intermediary. When a payment is made to a user, the LSP sends a notification to the NDS, configured with a specific webhook URL. The NDS then processes this information and dispatches a push notification to the intended mobile device, ensuring the user receives timely updates about incoming payments. This architecture necessitates vendors to set up and maintain their own NDS, tailored to handle and forward these notifications efficiently.
```
[the remark this answers is not in the document]
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+## Step 1: Run your own NDS
+As written above you will need to run your own NDS because NDS is configured to send push notifications to your app users, therefore configured with the required keys and certificates.
```
[the remark this answers is not in the document]
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+If you don't want to write it from scratch you can use our running <a href="https://github.com/breez/notify">NDS</a>
```
[the remark this answers is not in the document]
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+Our used NDS expects urls in the following format:
```
[the remark this answers is not in the document]
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+<section><i>https://your-nds-service.com/notify?platform=<ios|android>&token=[PUSH_TOKEN]</i></section>
+
+Once the NDS received such request it will send a push notification to the matched device.
```
[the remark this answers is not in the document]
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+* Wake Up the App: When a push notification is received, the NotificationServiceExtension will be triggered, waking up the app.
+
+* Connect with Breez SDK: Inside the extension, implement the logic to establish a connection with the Breez SDK which also runs the signer so the payment can be processed.
+
+* Wait for Payment Completion: Once connected, the app should wait for completion status from the Breez SDK that the payment has been received.
+
+* Handle the Notification: After confirming the payment, handle the notification accordingly by updating the app's UI.
+
+As a reference implelentation see how we did it in c-breez wallet: <a href="https://github.com/breez/c-breez/blob/main/ios/Breez%20Notification%20Service%20Extension/NotificationService.swift">NotificationService.swift</a>
```
[the remark this answers is not in the document]
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
+# Receiving Payment Notifications via Webhook in Breez SDK
+
+## Overview
+The Breez SDK provides a robust feature for applications to receive real-time notifications for incoming payments over the Lightning Network. This is achieved through the use of webhooks. A webhook is a powerful tool that allows your application to be notified via a specified URL when a payment is received.
+
+## Setting Up the Webhook
+To utilize this feature, you need to set up a webhook that Breez can call when a payment is received.
```
[the remark this answers is not in the document]
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+<section>
+
+```rust,ignore
+{{#include ../../snippets/rust/src/webhook.rs:register-webook}}
+```
+</section>
+</custom-tabs>
+
+## Handling Incoming Notifications
+When a payment is received, Breez will send a POST request to your webhook URL. The request body will contain details about the payment.
```
[the remark this answers is not in the document]
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md (hunk 1 above)
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md (hunk 2 above)
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md (hunk 3 above)
Comment (roeierez) [reply src/guide/payment_notification.md]: Done

File: src/guide/payment_notification.md (hunk 4 above)
Comment (roeierez) [reply src/guide/payment_notification.md]: I used different wording for now

File: snippets/rust/src/webhook.rs [hunk 15]
```diff
@@ -0,0 +1,21 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use breez_sdk_core::InputType::LnUrlWithdraw;
```
Comment (dangeross) [inline snippets/rust/src/webhook.rs:4]: This line is not needed. Maybe add `webhook` to `main.rs` so the CI picks it up.

**Review by dangeross — APPROVED** [review]
Looks good! Just needs the other languages.

**Review by ubbabeck — APPROVED** [review]
looks clear and concise.

**Pull request merged**